kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -148,6 +148,7 @@ def report_consumer_offsets_and_lag(self, consumer_offsets, highwater_offsets, c
                 self.log.warning(msg, consumer_group, topic, partition)
                 self.client.request_metadata_update()  # force metadata update on next poll()
         self.log.debug('%s consumer offsets reported', reported_contexts)
+        self.client.close_admin_client()
```

Comment:
It feels weird to close the client here in a sub-function. Couldn't close it at the very end of the check in the `check` function? (probably at the cost of a `try/finally` block

Reply:
Moved it one level up and also closed it in exception path.